staging/src/k8s.io/apimachinery/pkg/util/intstr/intstr.go

```diff
@@ -174,12 +174,17 @@ func getIntOrPercentValue(intOrStr *IntOrString) (int, bool, error) {
 	case Int:
 		return intOrStr.IntValue(), false, nil
 	case String:
-		s := strings.Replace(intOrStr.StrVal, "%", "", -1)
+		isPercent := false
+		s := intOrStr.StrVal
+		if strings.HasSuffix(s, "%") {
+			isPercent = true
+			s = strings.TrimSuffix(intOrStr.StrVal, "%")
+		}
 		v, err := strconv.Atoi(s)
 		if err != nil {
-			return 0, false, fmt.Errorf("invalid value %q: %v", intOrStr.StrVal, err)
+			return 0, isPercent, fmt.Errorf("invalid value %q: %v", intOrStr.StrVal, err)
 		}
-		return int(v), true, nil
+		return int(v), isPercent, nil
```

Comment:
Returning `isPercent ` here is a safe change for in-tree uses, since all the string values we pass to this method were already well-formatted percentages.

However, this is a breaking change for callers that passed in numeric strings (e.g. "10"). Previously, that would return `10, true, nil`. Now it returns `10, false, nil`.

Searching usages (https://grep.app/search?q=.GetValueFromIntOrPercent%28&filter[lang][0]=Go) this method is called externally.

If someone had documented their API to mean "integers are treated as absolute values, strings are treated as percentages", this change will break them.